mlrun/runtimes/databricks/databricks.py

```diff
@@ -18,32 +18,45 @@
 from mlrun.model import RunObject
 from mlrun.runtimes.kubejob import KubejobRuntime
 
+NOT_EXIST_VALUE = "handler_not_exist"
+
 
 class DatabricksRuntime(KubejobRuntime):
     kind = "databricks"
     _is_remote = True
 
-    def get_internal_code(self, runobj: RunObject):
+    def get_internal_parameters(self, runobj: RunObject):
         """
         Return the internal function code.
         """
+        internal_handler = runobj.spec.parameters.get("task_parameters", {}).get(
+            "internal_handler", NOT_EXIST_VALUE
+        )
+        # None and '' can be a symbol for none internal handler and don't need to be replaced.
+        if internal_handler == NOT_EXIST_VALUE:
+            internal_handler = runobj.spec.handler or ""
         encoded_code = (
             self.spec.build.functionSourceCode if hasattr(self.spec, "build") else None
         )
+        if not encoded_code:
+            return ""
         decoded_code = b64decode(encoded_code).decode("utf-8")
         code = _databricks_script_code + decoded_code
-        if runobj.spec.handler:
-            code += f"\n{runobj.spec.handler}(**handler_arguments)\n"
+        if internal_handler:
+            code += f"\n{internal_handler}(**handler_arguments)\n"
         code = b64encode(code.encode("utf-8")).decode("utf-8")
-        return code
+        return code, internal_handler
 
     def _pre_run(self, runspec: RunObject, execution):
-        internal_code = self.get_internal_code(runspec)
+        internal_code, internal_handler = self.get_internal_parameters(runspec)
         if internal_code:
             task_parameters = runspec.spec.parameters.get("task_parameters", {})
-            task_parameters["spark_app_code"] = self.get_internal_code(runspec)
+            task_parameters["spark_app_code"] = internal_code
+            if internal_handler:
+                task_parameters[
+                    "internal_handler"
+                ] = internal_handler  # in order to handle reruns.
```

Comment:
I don't think this comment is instructive, because it's not really clear what this code has to do with reruns. As far as I can tell, it's not so much that this solution supports reruns as it is that the previous solution was breaking them.